openghg/retrieve/_search.py

```diff
@@ -243,46 +243,55 @@ def search_footprints(
         period: Period of measurements. Only needed if this can not be inferred from the time coords
         continuous: Whether time stamps have to be continuous.
         retrieve_met: Whether to also download meterological data for this footprints area
-        high_spatial_res : Indicate footprints include both a low and high spatial resolution.
-        high_time_res: Indicate footprints are high time resolution (include H_back dimension)
+        high_spatial_resolution : Indicate footprints include both a low and high spatial resolution.
+        high_time_resolution: Indicate footprints are high time resolution (include H_back dimension)
                         Note this will be set to True automatically if species="co2" (Carbon Dioxide).
         short_lifetime: Indicate footprint is for a short-lived species. Needs species input.
                         Note this will be set to True if species has an associated lifetime.
         kwargs: Additional search terms
     Returns:
         SearchResults: SearchResults object
     """
+    # NOTE: the following two lines are a shorter way to get args...
+    # args = locals().copy()  # get dict of arguments to process and pass to `search`
+    # args.pop("kwargs")  # the value of kwargs is a dictionary, so we remove it
```

Comment:
You can try including this if you prefer and check the tests pass?

Reply:
This made some tests fail, I'm not really sure why. I think a better model (to implement later) would be to have metadata objects that we pass between internal functions, and have the wrapper functions like `standardise_footprint` populate the metadata objects. Then `search_footprints` might just be creating a metadata object (which does formatting), and passing the data type and metadata object to `search'. (Or we could have a 'FootprintsMetadata' child class...) I guess the way I rewrote the function is a bit more like that logic, but I can revert it to the original version if you like.

Reply:
I'm fine with how you rewrote the function below - your approach below seems great to me.